tests/foreman/rhai/test_rhai.py

```diff
@@ -54,54 +46,37 @@ def test_rhai_client_setup(self):
                     'subscription_id': subscription['id'],
                 })
                 break
+        cls.org_label = org.label
+        cls.ak_name = activation_key.name
+        cls.org_name = org.name
 
-        # Create VM
-        with VirtualMachine(distro='rhel66') as vm:
-            # Download and Install ketello-ca rpm
-            vm.install_katello_cert()
-            vm.register_contenthost(activation_key.name, org.label)
-
-            # Red Hat Access Insights requires RHEL 6/7 repo and it not
-            # possible to sync the repo during the tests,
-            # adding a file in /etc/yum.repos.d/rhel6/7.repo
-
-            rhel6_repo = conf.properties.get('insights.rhel6_repo')
+        super(RHAITestCase, cls).setUpClass()
 
-            repo_file = (
-                '[rhel6-rpms]\n'
-                'name=RHEL6\n'
-                'baseurl={0}\n'
-                'enabled=1\n'
-                .format(rhel6_repo)
-            )
-
-            vm.run(
-                'echo "{0}" >> /etc/yum.repos.d/rhel6.repo'
-                .format(repo_file)
-            )
+    def test_client_registration_to_rhai(self):
+        """@Test: Check for client registration to redhat-access-insights
+        service.
 
-            # Install redhat-access-insights package
-            package_name = 'redhat-access-insights'
-            result = vm.run('yum install -y {0}'.format(package_name))
-            self.assertEqual(result.return_code, 0)
+        @Feature: RHEL client registration to rhai
 
-            # Verify if package is installed by query it
-            result = vm.run('rpm -q {0}'.format(package_name))
-            self.assertEqual(result.return_code, 0)
+        @Assert: Registered client should appear in the Systems sub-menu of Red
+        Hat Access Insights
 
-            # Register client with Red Hat Access Insights
-            result = vm.run('redhat-access-insights --register')
-            self.assertEqual(result.return_code, 0)
+        """
+        # Register a VM to Access Insights Service
+        self.rhai.register_client_to_rhai(
```

Comment:
The vm will be destroyed when this method returns and will not allow you to do any assertions. That makes sense?